pkg/eval/options_test.go

```diff
@@ -1,28 +1,45 @@
 package eval
 
-import "testing"
+import (
+	"errors"
+	"testing"
+)
 
 type opts struct {
 	FooBar string
 	POSIX  bool `name:"posix"`
 	Min    int
+	ignore bool // this should be ignored since it isn't exported
 }
 
 var scanOptionsTests = []struct {
 	rawOpts  RawOptions
 	preScan  opts
 	postScan opts
+	err      error
 }{
 	{RawOptions{"foo-bar": "lorem ipsum"},
-		opts{}, opts{FooBar: "lorem ipsum"}},
+		opts{}, opts{FooBar: "lorem ipsum"}, nil},
 	{RawOptions{"posix": true},
-		opts{}, opts{POSIX: true}},
+		opts{}, opts{POSIX: true}, nil},
+	// Since "ignore" is not exported it will result in an error when used.
+	{RawOptions{"ignore": true},
+		opts{}, opts{ignore: false}, errors.New("unknown option ignore")},
 }
 
 func TestScanOptions(t *testing.T) {
+	// scanOptions requires a pointer to struct.
+	err := scanOptions(RawOptions{}, opts{})
+	if err == nil {
+		t.Errorf("Scan should have reported invalid arg")
+	}
+
 	for _, test := range scanOptionsTests {
 		opts := test.preScan
-		scanOptions(test.rawOpts, &opts)
+		err := scanOptions(test.rawOpts, &opts)
+		if err != nil && test.err == nil {
+			t.Errorf("Scan %v => %v, unexpected error %v", test.rawOpts, opts, err)
+		}
```

Comment:
* This isn't checking against the actual value of `test.err`. Either check that (you can check the error message), or just specify a boolean in the test struct.

* This doesn't cover the case where `err == nil && test.err != nil`.

Reply:
Yeah, I was taking an unwarranted shortcut based on the knowledge there is currently only one error condition. Fixed so the test is now future proof.